app/views/reports/report_average.html.erb

```diff
@@ -0,0 +1,78 @@
+<%= link_to "back", reports_path, class: "btn btn-info" %>
+
+<% 
```

Comment:
I recommend you to move all this logic to a different place, remember that the Views in MVC are only to show the information.

Check these links and let's talk in Slack if you want
https://stackoverflow.com/a/13179685
https://jtway.co/cleaning-up-your-rails-views-with-view-objects-42cf048ea491

Reply:
Fully agree with @jclopezdev